### PR TITLE
Add Subword Navigation

### PR DESCRIPTION
**Pull request by dmitsuki**
This commit adds subword navigation and allows them to be mapped to cursorSubWordRight, deleteSubWordRight and selectSubWordRight with left equivalents.

### CODE REVIEW

File: src/vs/editor/common/config/config.ts [hunk 1]
```diff
@@ -244,6 +245,14 @@ registerWordCommand(H.CursorWordRight, false, KeyCode.RightArrow);
 registerWordCommand(H.CursorWordRightSelect, true, KeyCode.RightArrow);
 registerWordCommand(H.DeleteWordLeft, false, KeyCode.Backspace);
 registerWordCommand(H.DeleteWordRight, false, KeyCode.Delete);
+registerWordCommand(H.DeleteSubWordLeft,false, KeyCode.F7);
+registerWordCommand(H.DeleteSubWordRight,false, KeyCode.F6);
```
Comment (SamVerschueren) [inline src/vs/editor/common/config/config.ts]: space between `,` and `false`

File: src/vs/editor/common/controller/oneCursor.ts [hunk 2]
```diff
@@ -583,6 +586,63 @@ export class OneCursorOp {
 		return true;
 	}
 
+	public static moveSubWordLeft(cursor:OneCursor, inSelectionMode: boolean, ctx: IOneCursorOperationContext): boolean {
+		let position = cursor.getPosition();
+		var lineNumber = position.lineNumber;
```
Comment (SamVerschueren) [inline src/vs/editor/common/controller/oneCursor.ts]: Use `const` as much as possible. If the value changes, like `positionFound` at L608, use `let`.

**Comment by alexandrudima** [comment]
This is quite a large change without tests.

Today we have the cut-off date for Dec 2015, so I don't feel confident enough in merging this in for the Dec 2015 release.

**Comment by kumarharsh** [comment]
@alexandrudima  This is a very useful feature, which I love in atom, makes editing all those camelCases in JS code a breeze.

**Comment by kumarharsh** [comment]
@alexandrudima can this be picked up in May iteration, pretty please :)

**Comment by alexandrudima** [comment]
@kumarharsh I cannot accept this PR because it doesn't have any tests (which makes it very difficult for me to maintain the code).

Given we're waiting since December, I don't think tests will be added.

**Comment by dmitsuki** [comment]
Sorry, didn't know I was supposed to add test. I have already switched to Sublime Text anyway. If somebody wants this feature they will have to write the test themselves.

**Pull request closed without merging**